### PR TITLE
American Lights 2011

### PR DESCRIPTION
**Pull request by ShiftyRail**
Maybe if I give it a less offensive name they won't try to crash the serb with it.
I didn't fix that lag machine bug, just did optimisation where I thought it would be needed.

Also made wall shadows less retarded, though they still look bad in many places.

### CODE REVIEW

**Comment by ShiftyRail** [comment]
I think this fails linter because I use map_override. Normally it should work fine for test server purposes.

**Comment by Jackreturnswithaveangenace** [comment]
Is the lag machine the teleporter thing

**Comment by ShiftyRail** [comment]
Yeah, among maybe other things

**Comment by SonixApache** [comment]
more importantly, is that lag machine fixed
also if this isn't ready to merge it should probably be drafted / marked with DNM

**Comment by ShiftyRail** [comment]
Lag machine isn't fixed yet because I dunno exactly the specifics (yet).

**Comment by Exxion** [comment]
>Maybe if I give it a less offensive name they won't try to crash the serb with it.

The lag machine was not made on purpose, for the record.

**Comment by ShiftyRail** [comment]
I know, I was being facetious.

**Comment by ShiftyRail** [comment]
Okay, that's lag machine fixed. I know why most shadows on wall look bad. The fix, however, is highly non-trivial so we're going to have to wait a bit.

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -1487,6 +1487,8 @@ Use this proc preferably at the end of an equipment loadout
 				for(var/atom/A in listed_turf)
 					if(!A.mouse_opacity && !A.name)
 						continue
+					if (istype(A, /atom/movable/light))
+						continue
```
Comment (Exxion) [inline code/modules/mob/mob.dm:1484]: Do you really need a special condition for these? Couldn't you just not give them names?

File: code/game/objects/effects/portals.dm [hunk 2]
```diff
@@ -159,6 +159,10 @@ var/list/portal_cache = list()
 /obj/effect/portal/proc/teleport(atom/movable/M as mob|obj)
 	if(istype(M, /obj/effect)) //sparks don't teleport
 		return
+	if(istype(M, /atom/movable/light))
+		var/atom/movable/light/L = M
+		if (istype(L.holder, /obj/effect)) // sparks lights don't teleport either
+			return
```
Comment (Exxion) [inline code/game/objects/effects/portals.dm:166]: Should light overlays be teleported by portals *at all*? Legitimate question; I don't know the code well enough to know if that's actually intended.
Though I will say that if it is intended, it means that objects with lights will produce extra sparks and louder sounds when going through portals.

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/modules/mob/mob.dm:1484]: well, it's useful to distinguish between the light obj and the "shadows over walls" one, but yeah.

File: code/game/objects/effects/portals.dm (hunk 2 above)
Comment (ShiftyRail) [reply code/game/objects/effects/portals.dm:166]: They should, yes. It ensure that the flashlight still casts light when going through the portal.

Comment (Exxion) [reply code/game/objects/effects/portals.dm:166]: Why wouldn't whatever causes the light to move normally also cause it to move with the flashlight through the portal?

Comment (ShiftyRail) [reply code/game/objects/effects/portals.dm:166]: ... yeah, that's already handled by `follow_holder`. I will fix it.

**Comment by KIBORG04** [comment]
When is merge?

**Comment by ShiftyRail** [comment]
Merge will happen when real life occurrences are sorted out. Don't worry, I
very much want this to happen but there are just other things going on.

File: code/modules/lighting/light_effect.dm [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+		set_dir(holder.loc.dir)
+	else
+		set_dir(holder.dir)
+
+// Moves the light overlay to the holder's turf and updates bleeding values accordingly.
+/atom/movable/light/proc/follow_holder()
+	for (var/turf/T in view(src, light_range))
+		if (T.shadow_atom == "\ref[src]")
+			T.shadow_atom = null
+	if(lighting_update_lights)
```
Comment (DamianX) [inline code/modules/lighting/light_effect.dm:96]: this is always true

**Comment by Eneocho** [comment]
@ShiftyRail Today I will remind em.

**Comment by ShiftyRail** [comment]
Conflicts should be fixed now, I'm trying to compile this thing and playtest it but DreamDeamon is not playing nice. Apologies for this

**Comment by Kurfursten** [comment]
Map render failing and new conflicts too

Feel free to ping me when you do get conflicts resolved and I will give this priority so we can do it live

**Comment by ShiftyRail** [comment]
Closing this and making a fresh start without the bugs that got me stuck.

**Pull request closed without merging**